File: frontend/src/core/utils/crypto.js
import CryptoJS from 'crypto-js'

const SECRET_KEY = import.meta.env.VITE_PASSWORD_ENCRYPT_KEY || ''
const ENABLE_ENCRYPT = !!SECRET_KEY

export function encryptPassword(password) {
  if (!password) return ''
  
  if (!ENABLE_ENCRYPT) {
    return password
  }
  
  const timestamp = Date.now().toString()
  const timeHex = timestamp.toString(16).padStart(16, '0')
  const iv = CryptoJS.enc.Utf8.parse(timeHex.substring(0, 16))
  
  const key = CryptoJS.enc.Utf8.parse(SECRET_KEY.padEnd(32, '0').substring(0, 32))
  
  const encrypted = CryptoJS.AES.encrypt(password, key, {
    iv: iv,
    mode: CryptoJS.mode.CBC,
    padding: CryptoJS.pad.Pkcs7
  })
  
  return timeHex + ':' + encrypted.ciphertext.toString(CryptoJS.enc.Base64)
}

export function decryptPassword(encryptedData) {
  if (!encryptedData || !encryptedData.includes(':')) {
    return encryptedData
  }
  
  if (!ENABLE_ENCRYPT) {
    return encryptedData
  }
  
  try {
    const [timeHex, encryptedBase64] = encryptedData.split(':')
    const iv = CryptoJS.enc.Utf8.parse(timeHex.substring(0, 16))
    
    const key = CryptoJS.enc.Utf8.parse(SECRET_KEY.padEnd(32, '0').substring(0, 32))
    
    const decrypted = CryptoJS.AES.decrypt(encryptedBase64, key, {
      iv: iv,
      mode: CryptoJS.mode.CBC,
      padding: CryptoJS.pad.Pkcs7
    })
    
    return decrypted.toString(CryptoJS.enc.Utf8)
  } catch (e) {
    return null
  }
}

export function isEncryptionEnabled() {
  return ENABLE_ENCRYPT
}

File: backend-v2/common/utils/crypto.py
"""
密码加密解密工具
"""

import os
import base64
from Crypto.Cipher import AES
from Crypto.Util.Padding import unpad


def decrypt_password(encrypted_data: str) -> str:
    """
    解密前端加密的密码
    
    前端加密格式: timeHex:encryptedBase64
    - timeHex: 16位十六进制时间戳
    - encryptedBase64: Base64编码的加密数据
    """
    if not encrypted_data or ':' not in encrypted_data:
        return encrypted_data
    
    secret_key = os.environ.get('PASSWORD_ENCRYPT_KEY', '')
    if not secret_key:
        return encrypted_data
    
    try:
        time_hex, encrypted_base64 = encrypted_data.split(':')
        
        iv = time_hex[:16].encode('utf-8')
        
        key = secret_key.ljust(32, '0')[:32].encode('utf-8')
        
        encrypted_bytes = base64.b64decode(encrypted_base64)
        
        cipher = AES.new(key, AES.MODE_CBC, iv)
        decrypted = unpad(cipher.decrypt(encrypted_bytes), AES.block_size)
        
        return decrypted.decode('utf-8')
    except Exception as e:
        return encrypted_data
